horace_core/sqw/@DnDBase/DnDBase.m

```diff
@@ -569,7 +569,11 @@
                 error('HORACE:DnDBase:invalid_argument',...
                     'input %s must be numeric array',field)
             end
-            obj.([field,'_']) = val;
+            if numel(val) == 1 && ~isempty(obj.([field,'_']))
+                obj.([field,'_'])(:) = val;
+            else
+                obj.([field,'_']) = val;
+            end
```

Comment:
a comment here saying what lines 572-6 do is needed

Reply:
Puzzled again. Clearly to process single values differently from arrays. And the code itself says how to process it. Will try to say something, but the code seems clearly states what it is doing. Its normally bad practice to describe in words what code is doing algorithmically. One should describe intentions.  And intention here is clear. Set up the value to the property.

Reply:
I can see that if field is an array then it will do what you say. But will this work if field is not an array, i.e. if (:) does not work on it?